DateTimeCalendar/App.java

```diff
@@ -0,0 +1,698 @@
+        expectedResult[29] = new DateTime(2007, Date.Month.APR, 8, 21, 51, 50); //0
+        expectedResult[30] = new DateTime(2013, Date.Month.MAR, 1, 7, 13, 10); //0
+        expectedResult[31] = new DateTime(2001, Date.Month.JUN, 16, 13, 31, 31); //0
+        expectedResult[32] = new DateTime(2011, Date.Month.MAR, 25, 22, 40, 40); //0
+        expectedResult[33] = new DateTime(2022, Date.Month.FEB, 5, 8, 50, 50); //0
+        expectedResult[34] = new DateTime(2043, Date.Month.OCT, 27, 5, 11, 10); //0
+        expectedResult[35] = new DateTime(2087, Date.Month.APR, 8, 21, 51, 50); //0
+        expectedResult[36] = new DateTime(1900, Date.Month.DEC, 32, 25, 61, 61);
+        expectedResult[37] = new DateTime(1900, Date.Month.DEC, 32, 25, 61, 61);
+        expectedResult[38] = new DateTime(1900, Date.Month.DEC, 32, 25, 61, 61);
+        expectedResult[39] = new DateTime(1900, Date.Month.DEC, 32, 25, 61, 61);
+        expectedResult[40] = new DateTime(1900, Date.Month.DEC, 32, 25, 61, 61);
+        expectedResult[41] = new DateTime(1900, Date.Month.DEC, 32, 25, 61, 61);
+        expectedResult[42] = new DateTime(1900, Date.Month.DEC, 32, 25, 61, 61);
+        
+        for (int i = 0; i < addBy.length; i++) {
+            boolean testPass = false;
+            int compareToExpected;
+
+            try {
+                DateTime actualResult = dateTimeAddControlObj.add(addBy[i][0], addBy[i][1], addBy[i][2], addBy[i][3], addBy[i][4], addBy[i][5]);
+                compareToExpected = expectedResult[i].compareTo(actualResult);
+
+                if (compareToExpected == 0) {
+                    testPass = true;
+                }
+
+                if (showOutput == 2) {
+                    System.out.println(String.format("Test DateTime.add() controlObj:%s modifier:%d/%d/%d %d:%d:%d // EO:%s AO:%s // TestPass:%s", dateTimeAddControlObj.toString(), addBy[i][0], addBy[i][1], addBy[i][2], addBy[i][3], addBy[i][4], addBy[i][5], expectedResult[i].toString(), actualResult.toString(), testPass ));
+                } else if (showOutput == 1) {
+                    if (testPass == false) {
+
+                    }
+                }
+
+                // System.out.println(actualResult.toString() + " " + actualResult.compareTo(expectedResult[i]));
+            } catch (IllegalArgumentException exception) {
+                if (expectedResult[i].compareTo(valueToCheckForException) == 0) {
+                    testPass = true;
+                } 
+
+                if (showOutput == 2) {
+                    System.out.println(String.format("Test DateTime.add() controlObj:%s modifier:%d/%d/%d %d:%d:%d // EO:%s AO:%s // TestPass:%s", dateTimeAddControlObj.toString(), addBy[i][0], addBy[i][1], addBy[i][2], addBy[i][3], addBy[i][4], addBy[i][5], expectedResult[i].toString(), "Invalid input", testPass ));
+                } else if (showOutput == 1) {
+                    if (testPass == false) {
```

Comment:
This has been corrected by using both the enum and by refactoring using some of the other points you have made for other changes. I set one of the expected results to something I knew would fail and am seeing a false response now.